### ISSUE TITLE
9999x9999 (and somewhat even 4k) is too slow to be usable

### ISSUE THREAD

**Issue by Equbuxu** [PixiEditor/PixiEditor#685]
https://github.com/user-attachments/assets/38496b5e-a73e-4d80-9f58-4b3084060f14

Feels like low res chunks aren't a thing anymore?

**Comment by flabbet** [comment]
Yeah at the moment it's not great, mostly because chunky image is not suited for new renderer. I want to implement ChunkyImage v2 with variable size chunks and proper gpu support, it should be good then

**Comment by flabbet** [comment]
Should be much better now, 4K seems to work very smooth for me, 9999x9999 still could be better (and will), but it's usable.
(2.0.0.46).

**Comment by wareya** [comment]
For me (ryzen 5 7600X, radeon rx 6800) a 8000x8000 canvas is basically just as slow as in OP's video, on current git master. I made sure to compile in Release mode instead of Debug.

**Comment by flabbet** [comment]
More optimizations are needed. Current renderer doesn't utilize chunk resolutions and all chunks are processed in full res.

**Comment by flabbet** [comment]
Implemented chunk rendering to whole graphs and scene in 2.0.0.92. Raster operations are still quite slow (dragging regions for example), but painting and vectors got significant performance improvements

**Comment by Lauloque** [comment]
Today on PixiEditor 2.0.1.19 (tested on both flatpak and snap packages) on my Kubuntu 25.10 with CPU AMD Ryzen 7 7700x, 64GB of DDR5 ram, GPU Nvidia RTX 4090:

I tried to edit a picture of 3580x1205 to "squarify" the picture:

- enlarge the canvas to a square that fits the original layer
- duplicate the layer
- make the bottom layer fill the square background
- blur the bottom layer

This was extremely laggy at every step, with frequent freezes of multiple seconds, especially when scaling the duplicate. And almost constant stutters of the GUI (might be hard to see in the screen recording, but you absolutely feel the gui being not responsive while you move the mouse around, click on things):

https://github.com/user-attachments/assets/dd234780-3ec9-46ca-8944-77c979d92741

In fact it lagged so much the video recording itself missed some frames. E.g. when I set a blur value and then went to saving, PixiEditor was stuttering a lot, but in the recording you just have a frame freeze of the recording from one to the other action.

And the resulting image, for some reason, is completely broken and not what I expected or even saw in the editor:

What i see in editor and expect as output:

<img width="367" height="431" alt="Image" src="https://github.com/user-attachments/assets/6aaab751-627f-43c1-ae5e-8a24a0b64301" />

What I get in the exported file:

<img width="932" height="956" alt="Image" src="https://github.com/user-attachments/assets/53480d28-1071-435e-8ac2-d99c40abbe69" />

**Comment by flabbet** [comment]
@Lauloque It could be, that PixiEditor doesn't use your RTX4090. You can check if it does in 2 ways:

1. run pixieditor from cli, it will display gpu information there
or
2. Go to Settings -> enable debug mode and type in seach bar "GPU". There will be an option to dump gpu info.

You may also want to switch to opengl (Render Api in the settings) and see if it helps.

Also, are you using proprietary nvidia drivers?

We made significant performance improvements in latest dev versions, you might want to check them out if nothing helps.

**Comment by Lauloque** [comment]
I am using proprietary Nvidia drivers:

```terminal
▶ nvidia-smi
Fri Feb 27 13:45:09 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 580.126.09             Driver Version: 580.126.09     CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090        Off |   00000000:01:00.0  On |                  Off |
|  0%   39C    P8             24W /  450W |    1708MiB /  24564MiB |     28%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
```

Running from CLI:

```terminal
▶ flatpak run net.pixieditor.PixiEditor
Running DrawieEngine with configuration:
	- RenderApi: Drawie.RenderApi.Vulkan.VulkanRenderApi
	- WindowingPlatform: 
	- DrawingBackend: Skia
	- Using GPU: NVIDIA GeForce RTX 4090 (NVIDIA)
```

I enabled debug and tried to run `Dump GPU Diagnostics` but then PixiEditor gets unresponsive to no end, had to kill it :')


I tried with openGL too, no better results in performance and output file:

```terminal
▶ Running DrawieEngine with configuration:
	- RenderApi: Drawie.RenderApi.OpenGL.OpenGlRenderApi
	- WindowingPlatform: 
	- DrawingBackend: Skia
	- Using GPU: NVIDIA GeForce RTX 4090/PCIe/SSE2 (NVIDIA Corporation)
```

Is there a place where I can try a dev package to see how it goes there?

**Comment by Mahri2D** [comment]
mine is very slow at 1024x1024 canvas especially if i use texture brush like "Foam".
I Tried Vulkan an OpenGL same result 

"Hardware Accelerated: True
Render API: Vulkan
GPU Info: AMD Radeon(TM) Vega 8 Graphics (AMD)
Details:
Device Type: IntegratedGpu
API Version: 4206842
Driver Version: 8388878"

**Comment by Lauloque** [comment]
Looks like both performance and incorrect export issues were fixed on my side as of PixiEditor v2.1.2.1 using the flathub package, on CachyOS.

<img width="1801" height="1209" alt="Image" src="https://github.com/user-attachments/assets/1feeb560-9d54-47cd-8eee-7b81350b17fd" />

Unsubscribing from this issue, ping me if needed.
